Jetpack Pro Dashboard: add atomic site actions to the dashboard site actions

yashwin opened the pull request:
Closes https://github.com/Automattic/jetpack-avalon/issues/45

## Proposed Changes

This PR implements adding atomic site actions(`Setup site`, `Change domain` & `Hosting configuration`) to the dashboard site actions.

### Testing Instructions

**Prerequisites**

Since these changes are made specifically for agencies, you must set yourself(partner) as an agency - 2c49b-pb. Make sure to switch it back to the previous type.

**Instructions**

1. Run `git checkout add/atomic-site-dashboard-site-actions` and `yarn start-jetpack-cloud`.
2. Open http://jetpack.cloud.localhost:3000/, and you'll be redirected to the /dashboard.
3. Verify that the previously added actions(`Issue new license`, `View activity`, `Copy this site`, `Site settings`, `View site` & `Visit WP Admin`) appear as expected and that the code changes make sense.
4. Since our current /sites endpoint does not include WPCOM atomic sites, we must modify the code to test and verify these changes.

<img width="339" alt="Screenshot 2023-08-31 at 3 32 12 PM" src="https://github.com/Automattic/wp-calypso/assets/10586875/09a74d10-10fa-4380-a2d8-4d70bbad071a">


Go to /jetpack-cloud/sections/agency-dashboard/sites-overview/site-actions/index.tsx#L42 and change the line to 

```
const isAtomicSite = true;
```

5. Verify that the atomic site actions(Setup site`, `Change domain` & `Hosting configuration`) are shown, and clicking on them redirects you to the correct links, and all open in a new tab. Also. verify that the event names are tracked when these menu items are clicked. 

<img width="313" alt="Screenshot 2023-08-31 at 3 32 56 PM" src="https://github.com/Automattic/wp-calypso/assets/10586875/103789ed-c9fa-42b5-9631-2e5f0ebd7123">


## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

yashwin committed:
Refactor site actions to an array and map through the array to show menu items

yashwin committed:
Add atomic site actions to the site action menu items

yashwin committed:
Add event names for atomic site actions

ilonagl reviewed (commented):
I am unsure how to test this, but it looks good from the screenshots design-wise. 👍

mattgawarecki commented on client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-actions/index.tsx:
```diff
@@ -65,58 +138,20 @@ export default function SiteActions( { isLargeScreen = false, site, siteError }:
-					</>
+				{ siteActions.map(
+					( action ) =>
+						action.isEnabled && (
```
Non-blocking:

If we're already doing a `.map`, this seems like the perfect opportunity to use `.filter` as well --

```js
siteActions
	.filter( ( { isEnabled } ) => isEnabled )
	.map( ... )
```

mattgawarecki commented on client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-actions/index.tsx:
```diff
@@ -38,12 +39,84 @@ export default function SiteActions( { isLargeScreen = false, site, siteError }:
+	const isWPCOMAtomicSiteCreationEnabled =
+		isEnabled( 'jetpack/pro-dashboard-wpcom-atomic-hosting' ) && isAtomicSite;
+
+	const siteActions = [
```
Non-blocking:

Considering the size of the array here, I could see reason for a new `useSiteActions` hook inside the `site-actions` directory. Thoughts?

mattgawarecki reviewed (approved):
All changes look reasonable to me, all unit tests pass, and functionally I saw the expected behavior when testing manually.

Approved with non-blocking comments, pending correction of a merge conflict in `types.ts`.

yashwin committed:
Refactor and create useSiteActions

yashwin commented:
@mattgawarecki: Thanks for your review. I totally agree with your thoughts and I have addressed your comments: https://github.com/Automattic/wp-calypso/pull/81263/commits/fffe57d52d75b50b2b35873317cc8645845cf149

- Created a new hook: useSiteActions and made some minor refactor
- Filtering the actions using `.filter` method

yashwin committed:
Use useMemo for actions

yashwin commented:
@mattgawarecki: Thanks for the review!

Apologies. I made some changes to the hook to use useMemo for actions. Do you mind having a quick look: https://github.com/Automattic/wp-calypso/pull/81263/commits/4a987ffecfdafc3b1ea06244a38c6d4e8c489a4d. Thanks!

mattgawarecki commented:
Still looking good 👍 Deploy at your convenience 🚢

yashwin commented:
Thanks, @mattgawarecki 🙇

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8870583

Thank you @yashwin for including a screenshot in the description! This is really helpful for our translators.
